### PR TITLE
zoom-us: 5.11.{1.8356,3.3882} -> 5.11.{9.10046,10.4400} 

### PR DESCRIPTION
**Pull request by kini**
###### Description of changes

Zoom started demanding I upgrade to >=5.11.9 the other day, so here's a bump. Someone will need to test this on x86_64-darwin and aarch64-darwin since I don't have access to those platforms - I just downloaded the appropriate files from the website and swapped in their hashes. The x86_64-linux version seems to work fine.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/instant-messengers/zoom-us/default.nix [hunk 1]
```diff
@@ -44,23 +44,23 @@ let
...
 
   # Zoom versions are released at different times for each platform
-  version = {
-    aarch64-darwin =import ./arm64-darwin-version.nix;
+  versions = {
+    aarch64-darwin = import ./arm64-darwin-version.nix;
     x86_64-darwin = import ./x86_64-darwin-version.nix;
     x86_64-linux = import ./x86_64-linux-version.nix;
-   }.${system} or throwSystem;
+  };
```
Comment (doronbehar) [inline pkgs/applications/networking/instant-messengers/zoom-us/default.nix:51]: Why did you change this?

File: pkgs/applications/networking/instant-messengers/zoom-us/default.nix [hunk 2]
```diff
@@ -109,7 +109,7 @@ let
 in
 stdenv.mkDerivation rec {
   pname = "zoom";
-  inherit version;
+  version = versions.${system} or throwSystem;
```
Comment (doronbehar) [inline pkgs/applications/networking/instant-messengers/zoom-us/default.nix:112]: Same here

**Review by doronbehar — COMMENTED** [review]
BTW are you aware you can use the `passthru.updateScript` with:

```
nix-shell maintainers/scripts/update.nix --argstr package zoom-us --argstr commit true
```

**Comment by kini** [comment]
>BTW are you aware you can use the passthru.updateScript with:

Unfortunately the update script doesn't work because Zoom has changed their page layout; for the Linux download page, the version number is not printed onto the screen until you click the dropdown menu and select "Other Linux". I figured I'd at least update the versions in a quick PR and fix the update script later. Also the update script seems to only update the version for my own platform (Linux) and not for other platforms. I figured Mac users would like an update too :)

File: pkgs/applications/networking/instant-messengers/zoom-us/default.nix (hunk 1 above)
Comment (kini) [reply pkgs/applications/networking/instant-messengers/zoom-us/default.nix:51]: I changed it so that all three attributes of `srcs` would have valid URLs. That helped me test that all three URLs are downloadable and have the right hash. Otherwise, two of the three attributes of `srcs` will be spurious URLs that may give 404 errors since they have the wrong version number for that platform (and in fact that is exactly what happens with the current state of the Zoom website).

File: pkgs/applications/networking/instant-messengers/zoom-us/default.nix (hunk 2 above)
Comment (kini) [reply pkgs/applications/networking/instant-messengers/zoom-us/default.nix:112]: Besides the reason I wrote above, evaluating `versions.${system}` down here is more consistent with how `srcs.${system}` is also evaluated down here inside the mkDerivation call rather than up in the let bindings.

**Comment by kini** [comment]
Ah good, it looks like they have a simple REST endpoint where we can get all the info we need ([[1]](https://zoom.us/rest/download?os=linux), [[2]](https://zoom.us/rest/download?os=mac)) without HTML-munging let alone JS-munging or Selenium or whatever... fixing update.sh should be easy too.

Should I make it update versions and hashes for all platforms, or is there a reason it's only supposed to do an update for the platform it's being run on?

**Comment by doronbehar** [comment]
> Should I make it update versions and hashes for all platforms, or is there a reason it's only supposed to do an update for the platform it's being run on?

Making the update script update all platforms at once would be great, but we can live without this feature. In any case, I still would like to make sure the update script works properly in this PR, at least for a single version.

**Comment by doronbehar** [comment]
So the update script worked as expected? Without any change?

**Comment by kini** [comment]
No, I need to fix it. I will push an update to the PR tomorrow. Thanks for the feedback so far :)

**Review by SuperSandro2000 — COMMENTED** [review]
Do we really need to split all the versions and hashes into one file each? If we need to split them into separate files at all why not use a json file?

**Comment by kini** [comment]
Agreed, I was already planning to do that when I saw your comment in fact :)

I've pushed an update which hopefully simplifies this satisfactorily. `update.sh` now updates all three platforms and can be run either directly or via the update.nix incantation @doronbehar mentioned, from any directory. In the meanwhile, Zoom has released yet another new version on Linux, so I bumped to that as well (5.11.10.4400).

**Review by doronbehar — APPROVED** [review]
Good Job!

**Comment by doronbehar** [comment]
Thanks for working on the update script.
